src/dune_rules/coq_rules.ml

```diff
@@ -77,11 +77,24 @@ module Bootstrap = struct
       | false -> Bootstrap lib
       | true -> Bootstrap_prelude)
 
-  let flags =
+  let boot_lib_flags ~coqdoc lib =
+    let open Command in
+    let dir = Coq_lib.src_root lib in
+    if coqdoc then
+      [ Args.A "--coqlib"; Args.Path (Path.build dir) ]
```

Comment:
What's the harm in passing --coqlib for everything?

Reply:
Good question, there are not exactly the same; I do prefer to handle at the coqc level the Coq library as a regular library. I don't like that special case.

coqdoc is a bit special as it needs to find the template files.

Reply:
For example, if -coqlib is set, it will try to do some magic for the plugins. I'd rather have the less magic possible happening when building with dune.

Reply:
This is what I meant by quirky dependencies of coqdoc earlier. The fact that coqlib is allowing it to find template files is silly. There are in fact arguments we can pass directly to coqdoc for these we might as well try.

Reply:
Yes. I'd say though "the dependency resolution method is quirky", as indeed the dependency on the style files is itself reasonable?

Or should coqdoc embed the templates so it can just generate them?

Reply:
I agree we should improve this, but not so clear for me how yet.

Reply:
For now I'll keep it like this, as anyways we seem to have some other bug in the way boot is resolved for coqdoc rules.